### DISCUSSION TITLE
Initialize Time Picker in the XAML code of a .NET MAUI app with the current time

### DISCUSSION THREAD

**Discussion by champ3615** [dotnet/maui#19544] [Q&A]
Hello experts, 

I would like to initialize the time picker in the XAML code of a .NET MAUI app with the current time. 

To do this, I've declared the namespace like this

xmlns:sys="clr-namespace:System; assembly=mscorlib"

and added the following XAML code

<TimePicker Time="{x:Static sys:DateTime.Now.TimeOfDay}" />

When I build the app, I get the following error message: 

XamlC error XFC0000: Cannot resolve type "clr-namespace:System; assembly=mscorlib:DateTime.Now"

What am I doing wrong?

By the way, if I initialize a date picker with the current date instead of the time picker, then this works:

<DatePicker Date="{x:Static sys:DateTime.Today}" />

Best regards,

Martin

**Comment by mohachouch** [comment]
Hello,

I think you use MVVM pattern. You can bind the Time property (Time span) and assign the value with the current time. If you don't use MVVM you can initialize the Time property in code behind.

<TimePicker Time="{Binding SelectedTime}" />

**Reply by champ3615** [reply]
Hi Mohamed,
Thanks for your reply. I must admit I'm just starting to learn how to write apps.
Could you please write the necessary XAML and C# code, so I know how I have to change the code in my app? That would be great?
Thanks!
Martin

**Reply by mohachouch** [reply]
Voici un exemple :

// Classe de modèle (ViewModel)
public class MainPageViewModel : INotifyPropertyChanged
{
    private TimeSpan _currentTime;

    public MainPageViewModel()
    {
        // Affecte la date courante à la propriété CurrentTime
        CurrentTime = DateTime.Now.TimeOfDay;
    }

    public TimeSpan CurrentTime
    {
        get { return _currentTime; }
        set
        {
            if (_currentTime != value)
            {
                _currentTime = value;
                OnPropertyChanged(nameof(CurrentTime));
            }
        }
    }

    // Le reste du code reste inchangé...
}



<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="VotreNamespace.MainPage">

    <StackLayout>
        
        <TimePicker x:Name="timePicker" Format="T" Time="{Binding CurrentTime}" />
    </StackLayout>

</ContentPage>

// Code-behind (dans le fichier .xaml.cs)
public partial class MainPage : ContentPage
{
    public MainPage()
    {
        InitializeComponent();
        BindingContext = new MainPageViewModel();
    }
}
